logistics_project/apps/malawi/warehouse/report_views/consumption_profiles.py

```diff
@@ -60,60 +62,118 @@ def consumption_row(sp, p, datespan):
 
 class View(warehouse_view.DistrictOnlyView):
 
-    def custom_context(self, request):
-        
-        sp = SupplyPoint.objects.get(location=request.location) \
-            if request.location else get_default_supply_point(request.user)
-        
-        table_headers = ["Product", "Total Actual Consumption for selected period",
-                         "# stockout days for all HSAs", 
-                         "Total consumption adjusted for stockouts", 
-                         "Data coverage (% of period)", 
-                         "Total consumption (adjusted  for stockouts and data coverage)", 
-                         "AMC for all HSAs"]
-        
-        hsa_list = selected_hsa = hsa_table = None
-        
-        if is_country(sp):
-            type = "national"
-        elif is_district(sp):
-            type = "district"
-        else:
-            assert is_facility(sp)
-            type = "facility"
-            hsa_list = hsa_supply_points_below(sp.location)
-            hsa_id = request.GET.get("hsa", "")
-            if hsa_id:
-                selected_hsa = SupplyPoint.objects.get(code=hsa_id) 
-                hsa_table = {
-                    "id": "hsa-consumption-profiles",
-                    "is_datatable": False,
-                    "is_downloadable": True,
-                    "header": table_headers,
-                    "data": [consumption_row(selected_hsa, p, request.datespan) for p in Product.objects.all()]
-                }
-            
-        
-        l_table = {
-            "id": "location-consumption-profiles",
-            "location_type": type,
+    automatically_adjust_datespan = True
+
+    def get_consumption_profile_table_headers(self, request):
+        base_level_description_plural = config.BaseLevel.get_base_level_description(request.base_level, plural=True)
+        return [
+            "Product",
+            "Total Actual Consumption for selected period",
+            "# stockout days for all %s" % base_level_description_plural,
+            "Total consumption adjusted for stockouts",
+            "Data coverage (% of period)",
+            "Total consumption (adjusted  for stockouts and data coverage)",
+            "AMC for all %s" % base_level_description_plural,
+        ]
+
+    def get_consumption_profile_table(self, request, supply_point, extra_table_params=None):
+        extra_table_params = extra_table_params or {}
+        table = {
             "is_datatable": False,
             "is_downloadable": True,
-            "header": table_headers,
-            "data": [consumption_row(sp, p, request.datespan) for p in Product.objects.all()]            
+            "header": self.get_consumption_profile_table_headers(request),
+            "data": [
+                consumption_row(supply_point, p, request.datespan)
+                for p in Product.objects.filter(type__base_level=request.base_level)
```

Comment:
looks like this has always been showing inactive products in this report. is that intentional?